.github/workflows/build.yml

```diff
@@ -0,0 +1,46 @@
+name: test
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [macos-latest]
+
+    steps:
+      - name: XCode version
+        run: xcode-select -p
+
+      - name: Check out Git repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Cache for ccache
+        uses: actions/cache@v1
+        env:
+          cache-name: cache-ccache
+        with:
+          path: ~/.ccache # ccache cache files are stored in `~/.ccache` on Linux/macOS
+          key: ${{ runner.os }}-build-${{ env.cache-name }}
```

Comment:
If you want to run this on every PR, you should cache base since that's only updated occasionally. See my CircleCI config.

https://github.com/koreader/koreader/blob/c20ad8f5e9385dd18138a306b874c5b20302f299/.circleci/config.yml#L20-L26

Reply:
I think running this on tagged releases is enough for https://github.com/koreader/koreader/issues/6875. No idea how to do that, how artifacts work and, generally speaking, the work involved on each release. So I would be happy if you could take care of it and remain ignorant :)

Reply:
In that case you can add a condition like https://github.com/hellerbarde/stapler/commit/48d2dc7e7a683ca3c2b659c1762aa67661f7ad79 

For tags/releases only we don't need to cache base.

It can be slightly annoying to test, but it's not too much of a bother on your own fork.

Anyhow, I can also do that in a follow-up PR some other time.